dali/python/nvidia/dali/external_source.py

```diff
@@ -129,18 +137,30 @@ def _is_generator_function(x):
     call = getattr(x, "__call__", None)
     return _is_generator_function(call)
 
+def _cycle_enabled(cycle):
+    if cycle is None:
+        return False
+    if cycle == False or cycle == "no":
```

Comment:
```suggestion
    if cycle is None or cycle == False or cycle == "no":
```

Reply:
None is handled separately - I don't know if merging it would make the code more readable.